Src/PerformanceCollector/Unit.Tests/QuickPulse/QuickPulseTelemetryProcessorTests.cs

```diff
@@ -694,5 +710,300 @@ public void QuickPulseTelemetryProcessorDoesNotCollectFullTelemetryItemsWhenSwit
+
+            // ACT
+            var dependencyShort = new DependencyTelemetry(
+                new string('c', MaxFieldLength),
+                new string('c', MaxFieldLength),
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false) { Context = { InstrumentationKey = instrumentationKey } };
+
+            var dependencyLong = new DependencyTelemetry(
+                new string('c', MaxFieldLength + 1),
+                new string('c', MaxFieldLength + 1),
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false) { Context = { InstrumentationKey = instrumentationKey } };
+
+            // process in the opposite order to allow for an easier validation order
+            telemetryProcessor.Process(dependencyLong);
+            telemetryProcessor.Process(dependencyShort);
+
+            // ASSERT
+            Assert.AreEqual(
+                ((DependencyTelemetryDocument)accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToList()[0]).CommandName,
+                dependencyShort.CommandName);
+            Assert.AreEqual(
+                ((DependencyTelemetryDocument)accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToList()[1]).CommandName,
+                dependencyLong.CommandName.Substring(0, MaxFieldLength));
+        }
+
+        [TestMethod]
+        public void QuickPulseTelemetryProcessorTruncatesLongFullDependencyTelemetryItemName()
+        {
+            // ARRANGE
+            var accumulatorManager = new QuickPulseDataAccumulatorManager();
+            var telemetryProcessor = new QuickPulseTelemetryProcessor(new SimpleTelemetryProcessorSpy());
+            var instrumentationKey = "some ikey";
+            ((IQuickPulseTelemetryProcessor)telemetryProcessor).StartCollection(
+                accumulatorManager,
+                new Uri("http://microsoft.com"),
+                new TelemetryConfiguration() { InstrumentationKey = instrumentationKey });
+
+            // ACT
+            var dependencyShort = new DependencyTelemetry(
+                new string('c', MaxFieldLength),
+                new string('c', MaxFieldLength),
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false)
+            { Context = { InstrumentationKey = instrumentationKey } };
+
+            var dependencyLong = new DependencyTelemetry(
+                new string('c', MaxFieldLength + 1),
+                new string('c', MaxFieldLength + 1),
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false)
+            { Context = { InstrumentationKey = instrumentationKey } };
+
+            // process in the opposite order to allow for an easier validation order
+            telemetryProcessor.Process(dependencyLong);
+            telemetryProcessor.Process(dependencyShort);
```

Comment:
I've taken another look, and I believe you're talking about the beginning of each test (where we create the objects under test and initialize them). The issue I'm seeing is that should I refactor that into a method, I would not going anything since three objects would be outputs of such method, so readability would still be the same (while all flexibility is lost). If you see a good option for extracting repetitive initialization out, please let me know.